plugin/pkg/shoot/validator/admission_test.go

```diff
@@ -1562,6 +1562,303 @@ var _ = Describe("validator", func() {
+					resourceCPU1    = resource.MustParse("2")
+					resourceCPU2    = resource.MustParse("3")
+					resourceMemory1 = resource.MustParse("2Gi")
+					resourceMemory2 = resource.MustParse("3Gi")
+					kubeletConfig   *core.KubeletConfig
+				)
+
+				BeforeEach(func() {
+					worker = core.Worker{
+						Name: "worker-name-kc",
+						Machine: core.Machine{
+							Type: "machine-type-kc",
+							Image: &core.ShootMachineImage{
+								Name: validMachineImageName,
+							},
+						},
+						Minimum: 1,
+						Maximum: 1,
+						Volume: &core.Volume{
+							VolumeSize: "40Gi",
+							Type:       &volumeType,
+						},
+						Zones: []string{"europe-a"},
+						Kubernetes: &core.WorkerKubernetes{
+							Kubelet: &core.KubeletConfig{
+								KubeReserved: &core.KubeletConfigReserved{
+									CPU:    &resourceCPU1,
+									Memory: &resourceMemory1,
+								},
+								SystemReserved: &core.KubeletConfigReserved{
+									CPU:    &resourceCPU1,
+									Memory: &resourceMemory1,
+								},
+							},
+						},
+					}
+
+					machineType := core.MachineType{
+						Name:   "machine-type-kc",
+						CPU:    resource.MustParse("5"),
+						GPU:    resource.MustParse("0"),
+						Memory: resource.MustParse("5Gi"),
+					}
+
+					kubeletConfig = &core.KubeletConfig{
+						KubeReserved:   &core.KubeletConfigReserved{},
+						SystemReserved: &core.KubeletConfigReserved{},
+					}
+
+					cloudProfile.Spec.MachineTypes = append(cloudProfile.Spec.MachineTypes, machineType)
+
+					Expect(coreInformerFactory.Core().InternalVersion().Projects().Informer().GetStore().Add(&project)).To(Succeed())
+					Expect(coreInformerFactory.Core().InternalVersion().CloudProfiles().Informer().GetStore().Add(&cloudProfile)).To(Succeed())
+					Expect(coreInformerFactory.Core().InternalVersion().Seeds().Informer().GetStore().Add(&seed)).To(Succeed())
+				})
+
+				It("should allow creation of Shoot if reserved resources are less than allocatable resources", func() {
+					shoot.Spec.Provider.Workers = append(shoot.Spec.Provider.Workers, worker)
+
+					attrs := admission.NewAttributesRecord(&shoot, nil, core.Kind("Shoot").WithVersion("version"), shoot.Namespace, shoot.Name, core.Resource("shoots").WithVersion("version"), "", admission.Create, &metav1.CreateOptions{}, false, nil)
+
+					err := admissionHandler.Admit(context.TODO(), attrs, nil)
+
+					Expect(err).To(BeNil())
+				})
+
+				It("should not throw any errors if both global and worker kubeletConfigs are nil", func() {
+					worker.Kubernetes.Kubelet = nil
+					shoot.Spec.Provider.Workers = append(shoot.Spec.Provider.Workers, worker)
+
+					attrs := admission.NewAttributesRecord(&shoot, nil, core.Kind("Shoot").WithVersion("version"), shoot.Namespace, shoot.Name, core.Resource("shoots").WithVersion("version"), "", admission.Create, &metav1.CreateOptions{}, false, nil)
+
+					err := admissionHandler.Admit(context.TODO(), attrs, nil)
+
+					Expect(err).To(BeNil())
```

Comment:
Any reason to not use

```suggestion
					Expect(err).ToNot(HaveOccurred())
```

?